mlir/test/Dialect/MemRef/expand-strided-metadata.mlir

```diff
@@ -1,20 +1,19 @@
 // RUN: mlir-opt --expand-strided-metadata -split-input-file %s -o - | FileCheck %s
 
+
+// memref.extract_strided_metadata is not folded away because meta-data on the %base memref 
+// can change in a future pass and folding away the Op here will cause incorrect lowering.
```

Comment:
I don't understand why the metadata could magically change? If they are static in the IR they shouldn't be able to just "change" out of nowhere.

Reply:
I'm also wondering about that. The ownership-based buffer deallocation pass uses memref.extract_strided_metadata a lot and removing this folding could make it less efficient (running time).

Reply:
Me too. I don't understand what is broken here and what we are fixing.